calyx-opt/src/passes/well_formed.rs

```diff
@@ -191,16 +191,24 @@ where
     Ok(())
 }
 
-fn same_type(proto_out: &CellType, proto_in: &CellType) -> CalyxResult<()> {
-    if proto_out != proto_in {
-        Err(Error::malformed_control(format!(
-            "Unexpected type for ref cell. Expected `{}`, received `{}`",
-            proto_out.surface_name().unwrap(),
-            proto_in.surface_name().unwrap(),
-        )))
-    } else {
-        Ok(())
+/// Returns true if cell_in is a subtype of the output cell.
```

Comment:
Maybe just a couple of sentences here about what we *do* currently check: every port is compatible, same width, etc.